Src/Workspaces/Core/Portable/Simplification/AbstractSimplificationService.cs

```diff
@@ -231,7 +231,7 @@ protected virtual SyntaxNode TransformReducedNode(SyntaxNode reducedNode, Syntax
             }, cancellationToken);
             }
 
-            Task.WaitAll(simplifyTasks, cancellationToken);
```

Comment:
Also, Task.WhenAll already handles cancellation, doesn't it?  I tried a quick and dirty experiment and i got a good OperationCanceledException out of it if any of the tasks were canceled.  Not sure what if hte WithCancellation bit is needed.